### PR TITLE
Show by default will show a summary in tabular format.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/136/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/137/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/138/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/139/
Test PASSed.

File: cmd/charm/charmcmd/show.go [hunk 1]
```diff
@@ -172,3 +196,125 @@ func handleIncludes(includes []string) (bool, []string, []string) {
...
+		sd.subordinate = charmMetadata["Subordinate"].(bool)
+	}
+	if _, ok := metadada["bundle-metadata"]; ok {
+		sd.bundle = true
+	}
+	return sd
+}
+
+func (s *showData) formatTabular() {
+	fmt.Fprintf(s.tw, "%s\t%s", "Name", s.name)
```
Comment (mhilton) [inline cmd/charm/charmcmd/show.go]: Why not "%s\t%s\n" and then you don't need the following Println?

File: cmd/charm/charmcmd/show.go [hunk 2]
```diff
@@ -172,3 +196,125 @@ func handleIncludes(includes []string) (bool, []string, []string) {
...
+	if len(s.terms) > 0 {
+		fmt.Fprintf(s.tw, "Terms\t%s", strings.Join(s.terms, ", "))
+		fmt.Fprintln(s.tw)
+	}
+	s.printChannels()
+}
+
+func (s *showData) printChannels() {
+	fmt.Fprintln(s.tw, " \t ")
+	fmt.Fprint(s.tw, "CHANNEL\tCURRENT")
```
Comment (mhilton) [inline cmd/charm/charmcmd/show.go]: Why is this one not fmt.Println?

File: cmd/charm/charmcmd/show.go [hunk 3]
```diff
@@ -172,3 +196,125 @@ func handleIncludes(includes []string) (bool, []string, []string) {
...
+		fmt.Fprintf(s.tw, "Tags\t%s", strings.Join(s.tags, ", "))
+		fmt.Fprintln(s.tw)
+		fmt.Fprintf(s.tw, "%s\t%t", "Subordinate", s.subordinate)
+		fmt.Fprintln(s.tw)
+	}
+}
+
+func toStringArray(a []interface{}) []string {
+	b := make([]string, len(a))
+	for i := range b {
```
Comment (mhilton) [inline cmd/charm/charmcmd/show.go]: I'd expect

``` go
for i, s := range a {
    b[i] = s.(string)
}
```

**Comment by mhilton** [comment]
:+1: with a couple of small thoughts

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/140/
Test PASSed.

File: cmd/charm/charmcmd/show.go (hunk 2 above)
Comment (fabricematrat) [reply cmd/charm/charmcmd/show.go]: we would like one more space, if I just do println it will be a new paragrah and not aligned with the first group.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/141/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/142/
Test PASSed.

**Comment by urosj** [comment]
Shipit.

**Comment by fabricematrat** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-client-merge

**Pull request merged**